examples/Precipitation_Map.py

```diff
@@ -0,0 +1,90 @@
+"""
+=====================
+NWS Precipitation Map
+=====================
+
+Plot a 1-day precipitation map using a netCDF file from the National Weather Service.
+
+This opens the data directly in memory using the support in the netCDF library to open from an
+existing memory buffer. In addition to CartoPy and Matplotlib, this uses a colortable from MetPy
+as well as MetPy's unit support.
+"""
+###############################
+# Imports
+from urllib.request import urlopen
+
+import cartopy.crs as ccrs
+import cartopy.feature as cfeat
+from metpy.plots import ctables
+from metpy.units import units, masked_array
+import matplotlib.colors as mcolors
+import matplotlib.pyplot as plt
+from netCDF4 import Dataset
+
+###############################
+# Download the data from the National Weather Service.
+data = urlopen('http://water.weather.gov/precip/downloads/2018/01/24/nws_precip_1day_20180124_conus.nc').read()
+Dataset('temp.nc', 'w').close()  # Work around bug where it needs an existing netCDF file
+nc = Dataset('temp.nc', 'r', memory=data)
+
+###############################
+# Pull the needed information out of the netCDF file
+prcpvar = nc.variables['observation']
+data = masked_array(prcpvar[:], units(prcpvar.units.lower())).to('mm')
+x = nc.variables['x'][:]
+y = nc.variables['y'][:]
+proj_var = nc.variables['polar_stereographic']
+
+###############################
+# Set up the projection information within CartoPy
+globe = ccrs.Globe(semimajor_axis=proj_var.earth_radius)
+proj = ccrs.Stereographic(central_latitude=90.0, central_longitude=proj_var.straight_vertical_longitude_from_pole,
+                          true_scale_latitude=proj_var.standard_parallel, globe=globe)
+
+###############################
+# Create the figure and plot the data
+# create figure and axes instances
+fig = plt.figure(figsize=(8, 8))
+ax = fig.add_subplot(1, 1, 1, projection=proj)
+
+# draw coastlines, state and country boundaries, edge of map.
+ax.coastlines()
+ax.add_feature(cfeat.BORDERS)
+
+# draw filled contours.
+clevs = [0, 1, 2.5, 5, 7.5, 10, 15, 20, 30, 40,
+         50, 70, 100, 150, 200, 250, 300, 400, 500, 600, 750]
+# In future MetPy
+# norm, cmap = ctables.registry.get_with_boundaries('precip', clevs)
```

Comment:
One other thing - shouldn't we spell it out `precipitation`?

Reply:
Sure.